src/Talis/Persona/Client/Users.php

```diff
@@ -0,0 +1,248 @@
+     * @throws \Exception
+     */
+    protected function personaGetUser($url, $token)
+    {
+        $curlOptions = array(
+            CURLOPT_URL             => $url,
+            CURLOPT_RETURNTRANSFER  => true,
+            CURLOPT_FOLLOWLOCATION  => true,
+            CURLOPT_TIMEOUT         => 30,
+            CURLOPT_HTTPHEADER      => array('Authorization: Bearer ' . $token)
+        );
+
+        $curl = curl_init();
+        curl_setopt_array($curl, $curlOptions);
+
+        $response = curl_exec($curl);
+        $headers = curl_getinfo($curl);
+        curl_close($curl);
+
+        if (isset($headers['http_code']) && $headers['http_code'] === 200)
+        {
+            return json_decode($response,true);
+        } else
+        {
+            throw new \Exception("Could not retrieve OAuth response code");
+        }
+    }
+
+    /**
+     * Create a new user in Persona
+     * @param string $url
+     * @param array $query
+     * @param string $token
+     * @throws \Exception
+     * @return array
+     */
+    protected function personaPostUser($url, $query, $token)
+    {
+        $curlOptions = array(
+            CURLOPT_POST            => true,
+            CURLOPT_URL             => $url,
+            CURLOPT_FOLLOWLOCATION  => true,
+            CURLOPT_RETURNTRANSFER  => true,
+            CURLOPT_TIMEOUT         => 30,
+            CURLOPT_POSTFIELDS      => json_encode($query),
+            CURLOPT_HTTPHEADER      => array('Authorization: Bearer ' . $token)
+        );
+
+        $curl = curl_init();
+        curl_setopt_array($curl, $curlOptions);
+
+        $response = curl_exec($curl);
+        $headers = curl_getinfo($curl);
+        curl_close($curl);
+
+        if (isset($headers['http_code']) && $headers['http_code'] === 200)
+        {
+            return json_decode($response,true);
+        } else
+        {
+            throw new \Exception((isset($responseDecoded['error_description']) ? $responseDecoded['error_description'] : 'Could not retrieve OAuth response code'));
+        }
+    }
+
+    /**
+     * Patch a Persona user
+     * @param string $url
+     * @param array $query
+     * @param string $token
+     * @return mixed
+     * @throws \Exception
+     */
+    protected function personaPatchUser($url, $query, $token)
+    {
+        $curlOptions = array(
+            CURLOPT_CUSTOMREQUEST   => 'PUT',
+            CURLOPT_URL             => $url,
+            CURLOPT_FOLLOWLOCATION  => true,
+            CURLOPT_RETURNTRANSFER  => true,
+            CURLOPT_TIMEOUT         => 30,
+            CURLOPT_POSTFIELDS      => json_encode($query),
+            CURLOPT_HTTPHEADER      => array('Authorization: Bearer ' . $token)
+        );
+
+        $curl = curl_init();
+        curl_setopt_array($curl, $curlOptions);
+
+        $response = curl_exec($curl);
+        $headers = curl_getinfo($curl);
+        curl_close($curl);
+
+        $responseDecoded = json_decode($response,true);
```

Comment:
as we've not checked the http code at this point, could $response contain something that can't be json_decoded?